authors/apps/authentication/serializers.py

```diff
@@ -10,12 +9,60 @@ class RegistrationSerializer(serializers.ModelSerializer):
 
     # Ensure passwords are at least 8 characters long, no longer than 128
     # characters, and can not be read by the client.
+    # ensures that password is required and not blank
+    email = serializers.EmailField()
+    username = serializers.CharField()
     password = serializers.CharField(
-        max_length=128,
-        min_length=8,
-        write_only=True
+        max_length=16,
+        write_only=True,
+        required=True,
+        error_messages={
+            "required": "Password field is required",
+            "blank": "Password field cannot be empty",
+        }
     )
 
+    def validate_password(self, password):
+        """
+            validates that  password is longer than 8 characters
+            password is alphanumeric
+        """
+        if len(password) < 8:
+            raise serializers.ValidationError(
+                "Password should atleast be 8 characters.")
+        if not re.search(r'[0-9]', password) or not re.search(r'[a-z]', password):
```

Comment:
@3Nakajugo This seems not check for a special character within the password accepts leading and  trailing spaces in the password. Fix this issue here.
#### Screenshots
*leading spaces with in the password*
<img width="953" alt="Screen Shot 2019-04-29 at 08 25 45" src="https://user-images.githubusercontent.com/36948493/56877710-a1755b80-6a58-11e9-9299-c79d7b089f32.png">
*no special character with in the password*
<img width="952" alt="Screen Shot 2019-04-29 at 08 18 48" src="https://user-images.githubusercontent.com/36948493/56877720-b356fe80-6a58-11e9-864d-4de66e8ea6ff.png">

Reply:
@archibishop  should the special characters be included in the password or exclude them?

Reply:
@3Nakajugo You should include them.